Lib/tkinter/__init__.py

```diff
@@ -30,6 +30,85 @@
 tk.mainloop()
 """
 
+import tkinter.constants
+
+__all__base = [
+    "TclError",
+    "TkVersion",
+    "TclVersion",
+    "READABLE",
+    "WRITABLE",
+    "EXCEPTION",
+    "EventType",
+    "Event",
+    "NoDefaultRoot",
+    "Variable",
+    "StringVar",
+    "IntVar",
+    "DoubleVar",
+    "BooleanVar",
+    "getint",
+    "getdouble",
+    "getboolean",
+    "XView",
+    "YView",
+    "Wm",
+    "Tk",
+    "Tcl",
+    "Pack",
+    "Place",
+    "Grid",
+    "Widget",
+    "Toplevel",
+    "Button",
+    "Canvas",
+    "Checkbutton",
+    "Entry",
+    "Frame",
+    "Label",
+    "Listbox",
+    "Menu",
+    "Menubutton",
+    "Message",
+    "Radiobutton",
+    "Scale",
+    "Scrollbar",
+    "Text",
+    "OptionMenu",
+    "Image",
+    "PhotoImage",
+    "BitmapImage",
+    "image_names",
+    "image_types",
+    "Spinbox",
+    "LabelFrame",
+    "PanedWindow",
+]
+
+__all__constants = [item for item in dir(tkinter.constants) if not item.startswith("__")]
+
+__all__modules = [
+    "colorchooser",
+    "commondialog",
+    "constants",
+    "dialog",
+    "dnd",
+    "filedialog",
+    "font",
+    "messagebox",
+    "scrolledtext",
+    "simpledialog",
+    "tix",
+    "ttk",
+]
+
+__all__internals_used_in_submodules = [
+    "_cnfmerge",      # Used in tkinter.dialog
+    "_default_root",  # Used in tkinter.simpledialog
```

Comment:
`_default_root` is [already imported in tkinter.simpledialog](https://github.com/python/cpython/blob/7a131a8fcd0bd5db2956223dc6cf0cd94a07824f/Lib/tkinter/simpledialog.py#L278-L279) and `_cnfmerge` is [used in dialog.py](https://github.com/python/cpython/blob/7a131a8fcd0bd5db2956223dc6cf0cd94a07824f/Lib/tkinter/dialog.py#L16)

The change for `_cnfmerge` is straightforward: I just have to duplicate the statement

```python
try: _cnfmerge = _tkinter._cnfmerge
except AttributeError: pass
```

in dialog.py

For `_default_root` I don't know how to change it without affecting the behaviour of `_QueryDialog`. I can't leave `parent` as None if no parent is passed because then the line [`if parent.winfo_viewable():`](https://github.com/python/cpython/blob/7a131a8fcd0bd5db2956223dc6cf0cd94a07824f/Lib/tkinter/simpledialog.py#L153) will fail.

Any advice?

Reply:
No, it is not imported, but is accessed as an attribute of `tkinter`.

The difference is that its value is changed after importing. When you import the submodule, it is None. After creating a root widget it keeps a reference to it, which can be used as a default parent in `_QueryDialog`.

Reply:
Thanks for the clarification!